driver/gamepad.c

```
@@ -241,6 +241,18 @@ static int gip_gamepad_op_guide_button(struct gip_client *client, bool down)
 	return 0;
 }
 
+static int gip_gamepad_op_authenticated(struct gip_client *client, int err)
+{
+	if (!err)
+	{
+		struct gip_gamepad *gamepad = dev_get_drvdata(&client->dev);
+		err = gip_gamepad_init_input(gamepad);
+		if (err)
+			return err;
+	}
+	return 0;
```

This whole thing could be simplified:
```c
if (err)
	return 0;

struct gip_gamepad *gamepad = dev_get_drvdata(&client->dev);
return gip_gamepad_init_input(gamepad);
```

Or even use the function directly instead of assigning it first:
```c
return gip_gamepad_init_input(dev_get_drvdata(&client->dev));
```

---

You are right, I wrote this without noticing that gip_set_encryption_key() makes no use of the return value. I certainly can simplify this or change gip_set_encryption_key() to forward result up to auth.c which would log any error code instead of silently failing in the future if any error is met inside gip_gamepad_init_input().